Source/SIMPLib/Geometry/GeometryHelpers.cpp

```diff
@@ -34,6 +34,205 @@
+IGeometry::LengthUnit LengthUnit::StringToType(const QString& str)
+{
+  IGeometry::LengthUnit t = IGeometry::LengthUnit::Unknown;
+
+  if(str == SIMPL::Geometry::k_Yoctometer)
+    return IGeometry::LengthUnit::Yoctometer;
+  if(str == SIMPL::Geometry::k_Zeptometer)
+    return IGeometry::LengthUnit::Zeptometer;
+  if(str == SIMPL::Geometry::k_Attometer)
+    return IGeometry::LengthUnit::Attometer;
+  if(str == SIMPL::Geometry::k_Femtometer)
+    return IGeometry::LengthUnit::Femtometer;
+  if(str == SIMPL::Geometry::k_Picometer)
+    return IGeometry::LengthUnit::Picometer;
+  if(str == SIMPL::Geometry::k_Nanometer)
+    return IGeometry::LengthUnit::Nanometer;
+  if(str == SIMPL::Geometry::k_Micrometer)
+    return IGeometry::LengthUnit::Micrometer;
+  if(str == SIMPL::Geometry::k_Millimeter)
+    return IGeometry::LengthUnit::Millimeter;
+  if(str == SIMPL::Geometry::k_Centimeter)
+    return IGeometry::LengthUnit::Centimeter;
+  if(str == SIMPL::Geometry::k_Decimeter)
+    return IGeometry::LengthUnit::Decimeter;
+  if(str == SIMPL::Geometry::k_Meter)
+    return IGeometry::LengthUnit::Meter;
+  if(str == SIMPL::Geometry::k_Decameter)
+    return IGeometry::LengthUnit::Decameter;
+  if(str == SIMPL::Geometry::k_Hectometer)
+    return IGeometry::LengthUnit::Hectometer;
+  if(str == SIMPL::Geometry::k_Kilometer)
+    return IGeometry::LengthUnit::Kilometer;
+  if(str == SIMPL::Geometry::k_Megameter)
+    return IGeometry::LengthUnit::Megameter;
+  if(str == SIMPL::Geometry::k_Gigameter)
+    return IGeometry::LengthUnit::Gigameter;
+  if(str == SIMPL::Geometry::k_Terameter)
+    return IGeometry::LengthUnit::Terameter;
+  if(str == SIMPL::Geometry::k_Petameter)
+    return IGeometry::LengthUnit::Petameter;
+  if(str == SIMPL::Geometry::k_Exameter)
+    return IGeometry::LengthUnit::Exameter;
+  if(str == SIMPL::Geometry::k_Zettameter)
+    return IGeometry::LengthUnit::Zettameter;
+  if(str == SIMPL::Geometry::k_Yottameter)
+    return IGeometry::LengthUnit::Yottameter;
+  if(str == SIMPL::Geometry::k_Angstrom)
+    return IGeometry::LengthUnit::Angstrom;
+  if(str == SIMPL::Geometry::k_Mil)
+    return IGeometry::LengthUnit::Mil;
+  if(str == SIMPL::Geometry::k_Inch)
+    return IGeometry::LengthUnit::Inch;
+  if(str == SIMPL::Geometry::k_Foot)
+    return IGeometry::LengthUnit::Foot;
+  if(str == SIMPL::Geometry::k_Mile)
+    return IGeometry::LengthUnit::Mile;
+  if(str == SIMPL::Geometry::k_Fathom)
+    return IGeometry::LengthUnit::Fathom;
+  if(str == SIMPL::Geometry::k_Unspecified)
+    return IGeometry::LengthUnit::Unspecified;
+  if(str == SIMPL::Geometry::k_Unknown)
+    return IGeometry::LengthUnit::Unknown;
+
+  return t;
+}
+
+// -----------------------------------------------------------------------------
+//
+// -----------------------------------------------------------------------------
+QString LengthUnit::TypeToString(IGeometry::LengthUnit t)
+{
+  QString geomType;
+
+  if(t == IGeometry::LengthUnit::Yoctometer)
```

Comment:
A switch statement would  be shorter and catch any additional cases as warnings as they are introduced in the future as long as you don't include a default case.  It wouldn't work from string to enum, but it's wonderful when converting the other way.

Also, you can remove `QString geomType;`  The last line can simply read `return QString();` to make it obvious that the value was never set anywhere.

All together, this will shrink this TypeToString length to half of what it is now.

Reply:
Made the changes along with a slightly different design that works better.